docs/source/tracking.rst

```diff
@@ -76,6 +76,101 @@ There are different kinds of remote tracking URIs:
   `[Azure] <http://docs.microsoft.com/azure/databricks/applications/mlflow/access-hosted-tracking-server>`_, or :ref:`the quickstart <quickstart_logging_to_remote_server>` to
   easily get started with hosted MLflow on Databricks Community Edition.
 
+How Runs and Artifacts are Recorded
+===================================
+As mentioned above, MLflow runs can be recorded to local files, to a SQLAlchemy compatible database, or remotely to a tracking server. MLflow artifacts can be persisted to local files
+and a variety of remote file storage solutions. For storing runs and artifacts, MLflow uses two components for `storage <https://mlflow.org/docs/latest/tracking.html#storage>`_: backend store and artifact store. While the backend store persists
+MLflow entities (runs, parameters, metrics, tags, notes, metadata, etc), the artifact store persists artifacts
+(files, models, images, in-memory objects, or model summary, etc).
+
+The MLflow client can interface with a variety of `backend <https://mlflow.org/docs/latest/tracking.html#backend-stores>`_ and `artifact <https://mlflow.org/docs/latest/tracking.html#artifact-stores>`_ storage configurations.
+Here are four common configuration scenarios:
+
+Scenario 1: MLflow on localhost
+-------------------------------
+
+Many developers run MLflow on their local machine, where both the backend and artifact store share a directory
+on the local filesystem—``./mlruns``—as shown in the diagram. The MLflow client directly interfaces with an
+instance of a `FileStore` and `LocalArtifactRepository`.
+
+.. figure:: _static/images/scenario_1.png
+
+In this simple scenario, the MLflow client uses the following interfaces to record MLflow entities and artifacts:
+
+ * An instance of a `LocalArtifactRepository` (to store artifacts)
+ * An instance of a `FileStore` (to save MLflow entities)
```

Comment:
Can we add github links to `LocalArtifactRepository`, `FileStore`, etc (`S3ArtifactRepository` down below)?

Reply:
You read my mind. I was wondering if should. By down below, where did you mean? Should we include them as links
while we mention in each scenario, or should we add a paragraph at the end as such:

The [FileStore](https://github.com/mlflow/mlflow/blob/master/mlflow/store/tracking/file_store.py#L115), [RestoreStore](https://github.com/mlflow/mlflow/blob/master/mlflow/store/tracking/rest_store.py#L39), and [SQLAlchemyStore](https://github.com/mlflow/mlflow/blob/master/mlflow/store/tracking/sqlalchemy_store.py#L61) are concrete implementations of the abstract class [AbstractStrore](https://github.com/mlflow/mlflow/blob/master/mlflow/store/tracking/abstract_store.py), whereas the  [LocalArtifactRepository](https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/local_artifact_repo.py#L15) and [S3ArtifactRepository](https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/s3_artifact_repo.py#L14) are concrete implemetnations of the abstract class [ArtifactRepository](https://github.com/mlflow/mlflow/blob/master/mlflow/store/artifact/artifact_repo.py#L13)

Reply:
Added this after the note. Better place.